lib/rubocop/cop/rspec/contain_exactly.rb

```diff
@@ -0,0 +1,41 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Prefer `match_array` when matching array values.
+      #
+      # @example
+      #   # bad
+      #   it { is_expected.to contain_exactly(*array1, *array2) }
+      #
+      #   # good
+      #   it { is_expected.to match_array(array1 + array2) }
+      #
+      #   # good
+      #   it { is_expected.to contain_exactly(content, *array) }
+      class ContainExactly < Cop
+        MSG = 'Prefer `match_array` when matching array values'.freeze
+
+        def on_send(node)
+          _receiver, method_name, *args = *node
+          return unless method_name == :contain_exactly
+
+          args.all? do |child_node|
+            child_node.is_a?(Node) && child_node.type == :splat
+          end && add_offense(node, :expression, MSG)
```

Comment:
the `&& add_offence` part looks weird. IMO it will make for a better style to have
```
args.any? do |child_node|
  return unless child_node.is_a?(Node) && child_node.type == :splat
end
```
then you can just add_offence at the end of the method